sceptre/cli/drift.py

```diff
@@ -0,0 +1,108 @@
+import click
+from click import Context
+
+from sceptre.context import SceptreContext
+from sceptre.cli.helpers import (
+    catch_exceptions,
+    write
+)
+from sceptre.plan.plan import SceptrePlan
+
+
+BAD_STATUSES = [
+    "DETECTION_FAILED",
+    "TIMED_OUT"
+]
+
+
+@click.group(name="drift")
+def drift_group():
+    """
+    Commands for calling drift detection.
+    """
+    pass
+
+
+@drift_group.command(name="detect", short_help="Run detect stack drift on running stacks.")
+@click.argument("path")
+@click.pass_context
+@catch_exceptions
+def drift_detect(ctx: Context, path: str):
+    """
+    Detect stack drift and return stack drift status.
+
+    In the event that the stack does not exist, we return
+    a DetectionStatus and StackDriftStatus of STACK_DOES_NOT_EXIST.
+
+    In the event that drift detection times out, we return
+    a DetectionStatus and StackDriftStatus of TIMED_OUT.
+
+    The timeout is set at 5 minutes, a value that cannot be configured.
+
+    :param path: The path to execute the command on.
+    :type path: str
+    """
+    context = SceptreContext(
+        command_path=path,
+        project_path=ctx.obj.get("project_path"),
+        user_variables=ctx.obj.get("user_variables"),
+        options=ctx.obj.get("options"),
+        output_format=ctx.obj.get("output_format"),
+        ignore_dependencies=ctx.obj.get("ignore_dependencies")
+    )
+
+    plan = SceptrePlan(context)
+    responses = plan.drift_detect()
+
+    exit_status = 0
+    for stack, response in responses.items():
+        status = response["DetectionStatus"]
+        if status in BAD_STATUSES:
+            exit_status += 1
+        for key in ["Timestamp", "ResponseMetadata"]:
+            response.pop(key, None)
+        write({stack.external_name: response}, context.output_format)
+
+    exit(exit_status)
+
+
+@drift_group.command(name="show", short_help="Shows stack drift on running stacks.")
```

Comment:
i've never used drift detection so this might be a silly question.  is the idea that sceptre users need to run two separate commands,  `drift detect` and then run `drift show` to see the actual drift?  If so, I'm wondering why bother to run detect when what I really want to do is just see the actual drift?  Could the user just run `drift show prod/mystack.yalm` and let sceptre handle running those commands in order?

Reply:
That's how it worked originally but @jfalkenstein made me rewrite it! But in fairness I like the new implementation and Jon apparently has a need for the detect only. To be clear, `drift detect` returns the `StackDriftStatus` and `drift show` returns a list of `StackResourceDriftStatus`. The former can be derived from the latter but this arguably cleaner.

Reply:
Just to clarify here, the idea is not that 

> sceptre users need to run two separate commands, `drift detect` and then run `drift show` to see the actual drift?

Rather:

- `drift detect` will just run drift detection and return (a dict that includes) `StackDriftStatus`
- `drift show` will run drift detection and then resource drift detect for each resource returned and return a list of dicts that include `StackResourceDriftStatus`.

In my original implementation, I only had the `drift show` command (although I named it something else). I had originally reasoned that no one will want `drift detect` since its response can be derived from `drift show`. However, Jon seems to have a valid use case for `drift detect`.

Reply:
hmm. ok, sounds good to me.